lib/galaxy/model/store/ro_crate_utils.py

```diff
@@ -220,7 +220,77 @@ def _add_workflows(self, crate: ROCrate):
 
             crate.license = self.workflow.license or ""
             crate.mainEntity["name"] = self.workflow.name
-            crate.mainEntity["subjectOf"] = cwl_wf
+
+            # Adding the creator information
+            if hasattr(self.workflow, 'creator_metadata') and self.workflow.creator_metadata:
+                creators = self.workflow.creator_metadata
+                if creators and isinstance(creators, list) and len(creators) > 0:
+                    first_creator = creators[0]
+                    creator_entity = crate.add(
+                        ContextEntity(
+                            crate,
+                            first_creator.get('identifier', ''),  # Default to empty string if identifier is missing
+                            properties={
+                                "@type": "Person",
+                                "name": first_creator.get('name', ''),  # Default to empty string if name is missing
+                                "orcid": first_creator.get('identifier', ''),  # Assuming identifier as orcid, or adjust accordingly
+                            },
+                        )
+                    )
+                    crate.mainEntity.append_to("creator", creator_entity)
+
+            # Add CWL workflow entity if exists
+            crate.mainEntity["subjectOf"] = cwl_wf if cwl_wf else ""
+
+        # Add tools used in the workflow
+        self._add_tools(crate)
+
+    def _add_tools(self, crate: ROCrate):
+        tool_entities = []
+
+        # Iterate over each step in the workflow
+        for step in self.workflow.steps:
+            # Check if the step corresponds to a tool
+            if step.type == "tool":
+                tool_id = step.tool_id
+                tool_version = step.tool_version
+                tool_name = step.label or tool_id  # use label if available, fallback to tool_id
```

Comment:
I would just use the tool_id. The label often describes what the tool does in a particular context, but that has no defined meaning you can associate, so that's probably not helpful here.